src/Core/Models/Business/SubscriptionCreateOptions.cs

```diff
@@ -16,28 +16,20 @@ public class OrganizationSubscriptionOptionsBase : Stripe.SubscriptionCreateOpti
         };
         foreach (var plan in plans)
         {
+            AddPlanIdToSubscription(plan);
+            AddPremiumAccessAddon(premiumAccessAddon, plan);
```

Comment:
I'm actually not sure that `AddPremiumAccessAddon` applies to SM. SM plans don't have a `plan.StripePremiumAccessPlanId`, so this method will never do anything for SM plans. And for now, while SM is an addon to PM, I think that's OK.

I recommend moving this method call into the PM `case` block. (but not SM)